### PR TITLE
US13 // mongoDB - database setup

### PR DESCRIPTION
**Pull request by matschi3**
Complete database setup. have fun reviewing 🫠 

[US13](https://github.com/matschi3/capstone/issues/23)
[Deployment](https://capstone-git-feature-us13database-setup-matschi3.vercel.app/)

Its just the app as it was before, but much slower because of free-mongoAtlasDB.
So click a button and wait a second or two until the magic happens✌🏼

If you accidentally used up all parts or deleted them while testing 
`->` message me on Slack and i will reset the parts-db for you ❤️

### CODE REVIEW

File: pages/api/categories/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import dbConnect from "../../../db/connect.js";
+import Category from "../../../db/models/Category.js";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  // get all categories from the categories db
+  if (request.method === "GET") {
+    const categories = await Category.find({});
+    return response.status(200).json(categories);
+  }
```
Comment (dimitriosxmi) [inline pages/api/categories/index.js:10]: consider adding an error case here

File: pages/_app.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import GlobalStyle from "../styles.js";
+import { SWRConfig } from "swr";
+
+const fetcher = (url) => fetch(url).then((response) => response.json());
```
Comment (dimitriosxmi) [inline pages/_app.js:4]: consider adding an error handling here later on for safety precautions 😄

File: pages/api/items/[id].js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      return response.status(404).json({ status: "Item nicht gefunden." });
+    }
+    response.status(200).json(item);
+  }
+  // update id-specific item from the items db
+  if (request.method === "PUT") {
+    const editedItem = request.body;
+    await Item.findByIdAndUpdate(id, editedItem);
+    response.status(200).json({ status: "Item bearbeitet." });
+  }
```
Comment (dimitriosxmi) [inline pages/api/items/[id].js:20]: this one should also have an error case

File: pages/api/items/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export default async function handler(request, response) {
+  await dbConnect();
+  // get all items from the items db + populate parts
+  if (request.method === "GET") {
+    const items = await Item.find({}).populate({
+      path: "parts",
+      populate: { path: "category" },
+    });
+    return response.status(200).json(items);
+  }
```
Comment (dimitriosxmi) [inline pages/api/items/index.js:13]: also one error case here pretty please 👁️ U 👁️

File: pages/api/parts/[id].js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      return response.status(404).json({ status: "Teil nicht gefunden." });
+    }
+    response.status(200).json(part);
+  }
+  // update id-specific part from the parts db
+  if (request.method === "PUT") {
+    const editedPart = request.body;
+    await Part.findByIdAndUpdate(id, editedPart);
+    response.status(200).json({ status: "Teil bearbeitet." });
+  }
```
Comment (dimitriosxmi) [inline pages/api/parts/[id].js:20]: here too!! :D error case please

File: pages/api/parts/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import dbConnect from "../../../db/connect.js";
+import Part from "../../../db/models/Part.js";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  // get all parts from the parts db
+  if (request.method === "GET") {
+    const parts = await Part.find({}).populate("category");
+    return response.status(200).json(parts);
+  }
```
Comment (dimitriosxmi) [inline pages/api/parts/index.js:10]: also here an error case 😄

**Comment by matschi3** [comment]
Thanks @dimitriosxmi for your time on Sunday ✌🏼

// added error handling to all the backend files

**Review by MariaRiosNavarro — APPROVED** [review]

**Comment by MariaRiosNavarro** [comment]
Error Handling added, LGFM 👍🏽

**Pull request merged**